src/main/java/eu/ai4work/sws/service/SlidingDecisionService.java

```diff
@@ -0,0 +1,67 @@
+package eu.ai4work.sws.service;
+
+import eu.ai4work.sws.model.specific.CongestionLevel;
+import eu.ai4work.sws.model.specific.TruckPosition;
+import eu.ai4work.sws.model.generic.SlidingDecisionResult;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@Service
+@RequiredArgsConstructor
+public class SlidingDecisionService {
+    private final RuleEngineService ruleEngineService;
+    public Map<String, Object> getSlidingDecision(Map<String, Object> parameters) {
+        Map<String, Object> decisionResults = new HashMap<>();
+
+        int noOfTrucksInQueue = (int) parameters.getOrDefault("noOfTrucksInQueue", 0);
+        int positionOfTruckToBePrioritized = (int) parameters.getOrDefault("positionOfTruckToBePrioritized", 0);
+
+        CongestionLevel congestionLevel = determineCongestionLevel(noOfTrucksInQueue);
+        TruckPosition truckPositionZone = determineTruckPositionZone(positionOfTruckToBePrioritized, noOfTrucksInQueue);
+
+        SlidingDecisionResult decisionOutcome = ruleEngineService.applySlidingDecisionRules(congestionLevel, truckPositionZone);
+        decisionResults.put("slidingDecision", decisionOutcome);
+        decisionResults.put("description", decisionOutcome.getDisplayName());
+
+        return decisionResults;
+    }
+
+    /**
+     * Determines the congestion level based on the number of trucks in the queue.
+     *
+     * @param noOfTrucks The total number of trucks in the queue.
+     * @return The congestion level as a CongestionLevel enum.
+     */
+    private CongestionLevel determineCongestionLevel(int noOfTrucks) {
+        if (noOfTrucks <= 0) {
```

Comment:
This can be removed as it is implicitly covered by the `<=10` in the next if.

Reply:
Done in the commit https://github.com/AI4WORK-Project/sliding-work-sharing/pull/2/commits/6e63b8bce37097860fbe069304276275bb28ba83